src/util/observatory.ts

```diff
@@ -0,0 +1,63 @@
+import {BoaviztaCpuImpactModel} from '../lib';
+
+/**
+ * Pipeline for computing impacts based on observation.
+ */
+export class Observatory {
+  private observations: any;
+
+  /**
+   * Init observations object.
+   */
+  constructor(observations: any) {
+    this.observations = observations;
+  }
+
+  /**
+   * Apply `boavizta` model computation to observations.
+   */
+  private async boaviztaHandler(params: any) {
```

Comment:
I know this is draft but eventually there should be a separate modelFactory which is initialised when the graph is loaded and during computation like this returns an instance of a model given the name that you are requesting.

Reply:
Agree, I have got rid of specific handlers locally (not pushed yet), trying to have a unified interface for all models.